src/Microsoft.DocAsCode.Common/Git/GitUtility.cs

```diff
@@ -112,22 +123,47 @@ private static GitDetail GetFileDetailCore(string filePath)
         private static GitRepoInfo GetRepoInfoCore(string directory)
         {
             var repoRootPath = RunGitCommandAndGetFirstLine(directory, GetRepoRootCommand);
-            var localBranch = GetLocalBranchName(repoRootPath);
 
+            // The "docfx..". environment variable specifies the branch name to use.
+            var localBranch = Environment.GetEnvironmentVariable("DOCFX_SOURCE_BRANCH_NAME");
             string remoteBranch;
-            try
+
+            // Many build systems use a "detached head", which means that the normal git commands
+            // to get branch names do not work.  Thankfully, they set an environment variable.
+            if (string.IsNullOrEmpty(localBranch))
             {
-                remoteBranch = RunGitCommandAndGetFirstLine(repoRootPath, GetRemoteBranchCommand);
-                var index = remoteBranch.IndexOf('/');
-                if (index > 0)
+                var isDetached = "HEAD" == RunGitCommandAndGetFirstLine(repoRootPath, GetLocalBranchCommand);
```

Comment:
I think the addition of `GetLocalBranchName` is wrong.  It changes the semantics of both  `RepoInfo.LocalBranch` and `RepoInfo.RemoteInfo`.  Previously they held a git branch name.  Now they can also hold a `commit id`.

I want nothing to do with this.  Its **evil**.

Not sure what you mean by `"HEAD" == Run [](start = 32, length = 14)`